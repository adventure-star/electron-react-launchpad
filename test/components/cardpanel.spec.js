import React from 'react';
import ReactTestRenderer from 'react-test-renderer';

import CardPanel from '../../src/renderer/components/cardpanel.jsx'

function setupStandard() {
    const renderer = ReactTestRenderer.create(<CardPanel className='testClass'>Test card content</CardPanel>);
    const component = renderer.toJSON();

    return {
        component
    };
}

describe('CardPanel component', () => {
    it('should be of type div', () => {
        const { component } = setupStandard();
        expect(component.type).toBe('div');
    });

    it('should have only the classes "card-panel" and "testClass"', () => {
        const { component } = setupStandard();
        expect(component.props.className).toBe('card-panel testClass');
    });

    it('children should be the text "Test card content"', () => {
        const { component } = setupStandard();
        expect(component.children[0]).toBe('Test card content');
    });

    it('Matches/Renders a snapshot', () => {
        const { component } = setupStandard();
        expect(component).toMatchSnapshot();
    });
});
