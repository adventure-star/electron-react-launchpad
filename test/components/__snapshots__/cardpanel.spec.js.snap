// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardPanel component Matches/Renders a snapshot 1`] = `
<div
  className="card-panel testClass"
>
  Test card content
</div>
`;
